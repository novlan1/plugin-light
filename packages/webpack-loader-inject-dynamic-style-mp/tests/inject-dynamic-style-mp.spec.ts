import * as fs from 'fs';
import path from 'path';
import * as LoaderUtils from 'loader-utils';
import injectDynamicStyleMp from '../src/loader';


jest.mock('fs');

const dir = path.resolve(process.cwd(), 'src/local-component/ui/tip-match/tip-match-notice');

const source = '<style lang="scss" src="./css/base.scss"></style>';
const sourceWithoutCSSPath = '<style lang="scss" src="./css/vars.scss"></style>';

describe('injectDynamicStyleMp', () => {
  const oldEnv = process.env;

  beforeAll(() => {
    jest.spyOn(fs, 'readdirSync').mockReturnValue(['base.scss', 'dialog.scss', 'icon.scss'] as any);
  });

  beforeEach(() => {
    process.env = { ...oldEnv };
    process.env.UNI_PLATFORM = 'mp-weixin';
  });

  afterEach(() => {
    process.env = oldEnv;
    jest.restoreAllMocks();
  });

  it('should not be used on non-MP platform', () => {
    process.env.UNI_PLATFORM = 'h5';
    expect(injectDynamicStyleMp.call({ context: dir }, source)).toBe(source);
  });

  it('should not be used when failed to match CSS path', () => {
    expect(injectDynamicStyleMp.call({ context: dir }, sourceWithoutCSSPath)).toBe(sourceWithoutCSSPath);
  });

  it('should work correctly', () => {
    expect(injectDynamicStyleMp.call({ context: dir }, source)).toMatchSnapshot();
  });

  it('should set topElement correctly', () => {
    jest.spyOn(LoaderUtils, 'getOptions').mockReturnValue({ topElement: '#container' });
    expect(injectDynamicStyleMp.call({ context: dir }, source)).toMatchSnapshot();
  });
});
