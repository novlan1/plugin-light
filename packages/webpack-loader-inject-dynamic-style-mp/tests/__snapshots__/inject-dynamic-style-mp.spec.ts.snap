// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`injectDynamicStyleMp should set topElement correctly 1`] = `
"<style lang="scss" >
#container.tip-match-notice {
  
  &--type-dialog {
    @import './css/dialog.scss';
  }

  &--type-icon {
    @import './css/icon.scss';
  }
}
</style>"
`;

exports[`injectDynamicStyleMp should work correctly 1`] = `
"<style lang="scss" >
body.tip-match-notice {
  
  &--type-dialog {
    @import './css/dialog.scss';
  }

  &--type-icon {
    @import './css/icon.scss';
  }
}
</style>"
`;
